language-server/src/features/workspace.js

```diff
@@ -11,8 +11,22 @@ import {
 import { TextDocument } from "vscode-languageserver-textdocument";
 import { publish, publishAsync, subscribe } from "../pubsub.js";
 import { getSchemaDocument } from "./schema-documents.js";
-import { isSchema } from "./document-settings.js";
-
+import { getDocumentSettings, schemaFilePatterns } from "./document-settings.js";
+import picomatch from "picomatch";
+
+
+export const isMatchedFile = (uri, patterns) => {
+  patterns = patterns.map((pattern) => `**/${pattern}`);
+  const matchers = patterns.map((pattern) => {
+    return picomatch(pattern, {
+      noglobstar: false,
+      matchBase: false,
+      dot: true,
+      nonegate: true
+    });
+  });
+  return matchers.some((matcher) => matcher(uri));
+};
```

Comment:
I prefer to put utility functions like this toward the bottom of the code. The code most relevant to the main purpose of the file should come first and supporting functions come after. Since this is exported, I'd put it near the top, but below the default export.

Reply:
I'll change it's position to the preferred one.